app/models/admin/tag_filter.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+class Admin::TagFilter
+  KEYS = %i(
+    status
+    name
+    order
+  ).freeze
+
+  attr_reader :params
+
+  def initialize(params)
+    @params = params.to_h.symbolize_keys
+  end
+
+  def results
+    scope = Tag.reorder(nil)
+
+    params.each do |key, value|
+      next if key == :page
+
+      scope.merge!(scope_for(key, value)) if value.present?
+    end
+
+    scope
+  end
+
+  private
+
+  def scope_for(key, value)
+    case key
+    when :status
+      status_scope(value)
+    when :name
+      Tag.search_for(value.to_s.strip, params[:limit], params[:offset], exclude_unlistable: false)
+    when :order
+      order_scope(value)
+    else
+      raise Mastodon::InvalidParameterError, "Unknown filter: #{key}"
+    end
+  end
+
+  def status_scope(value)
+    case value.to_s
+    when 'reviewed'
+      Tag.reviewed
+    when 'review_requested'
+      Tag.pending_review
+    when 'unreviewed'
+      Tag.unreviewed
+    when 'trendable'
+      Tag.trendable
+    when 'not_trendable'
+      Tag.not_trendable
+    when 'usable'
+      Tag.usable
+    when 'not_usable'
+      Tag.not_usable
+    else
+      raise Mastodon::InvalidParameterError, "Unknown status: #{value}"
+    end
+  end
```

Comment:
It seems to me many of these filters are orthogonal with each other and thus could be split. But this can be changed later if we need to, this does not need to block the PR.